dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchIncludeFragment.java

```diff
@@ -323,52 +313,33 @@ public void search() {
             // going to assume that this is NOT a linked object, for now:
             solrSearchResult.setIsInTree(true);
             // (we'll review this later!)
-
-            if (solrSearchResult.getType() == SearchObjectType.DATAVERSES) {
-                dataverseDao.populateDvSearchCard(solrSearchResult);
-                
-                /*
-                Dataverses cannot be harvested yet.
-                if (isHarvestedDataverse(solrSearchResult.getEntityId())) {
-                    solrSearchResult.setHarvested(true);
-                }*/
-
-            } else if (solrSearchResult.getType() == SearchObjectType.DATASETS) {
-                datasetVersionService.populateDatasetSearchCard(solrSearchResult);
-
-                // @todo - the 3 lines below, should they be moved inside
-                // searchServiceBean.search()?
-                String deaccesssionReason = solrSearchResult.getDeaccessionReason();
-                if (deaccesssionReason != null) {
-                    solrSearchResult.setDescriptionNoSnippet(deaccesssionReason);
-                }
-
-            } else if (solrSearchResult.getType() == SearchObjectType.FILES) {
-                dataFileService.populateFileSearchCard(solrSearchResult);
-
-                /**
-                 * @todo: show DataTable variables
-                 */
-            }
         }
+        Map<SearchObjectType, List<SolrSearchResult>> results = searchResults.stream()
+                .collect(Collectors.groupingBy(SolrSearchResult::getType));
+        solrSearchResultsService.populateDataverseSearchCard(
+                results.getOrDefault(SearchObjectType.DATAVERSES, Collections.emptyList()));
+        solrSearchResultsService.populateDatasetSearchCard(
+                results.getOrDefault(SearchObjectType.DATASETS, Collections.emptyList()));
+        solrSearchResultsService.populateDatafileSearchCard(
+                results.getOrDefault(SearchObjectType.FILES, Collections.emptyList()));
```

Comment:
As we are doing this in SearchIncludeFragment - we can also change MyDataFragment